File: 2283-sort-even-and-odd-indices-independently/2283-sort-even-and-odd-indices-independently.py
class Solution(object):
    def sortEvenOdd(self, nums):
        evens = sorted(nums[0::2])   
        odds = sorted(nums[1::2], reverse=True) 
        res = []
        
        e = o = 0
        for i in range(len(nums)):
            if i % 2 == 0:
                res.append(evens[e])
                e += 1
            else:
                res.append(odds[o])
                o += 1
        return res   

File: 2283-sort-even-and-odd-indices-independently/2283-sort-even-and-odd-indices-independently.java
import java.util.*;

class Solution {
    public int[] sortEvenOdd(int[] nums) {
        List<Integer> evens = new ArrayList<>();
        List<Integer> odds = new ArrayList<>();
        
        // Separate even and odd index values
        for (int i = 0; i < nums.length; i++) {
            if (i % 2 == 0) {
                evens.add(nums[i]);
            } else {
                odds.add(nums[i]);
            }
        }
        
        // Sort even ascending, odd descending
        Collections.sort(evens); 
        Collections.sort(odds, Collections.reverseOrder());
        
        int[] res = new int[nums.length];
        int e = 0, o = 0;
        
        // Merge back
        for (int i = 0; i < nums.length; i++) {
            if (i % 2 == 0) {
                res[i] = evens.get(e++);
            } else {
                res[i] = odds.get(o++);
            }
        }
        
        return res;
    }
}